x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/ClassificationTests.java

```diff
@@ -89,6 +89,7 @@ protected NamedWriteableRegistry getNamedWriteableRegistry() {
         return new NamedWriteableRegistry(entries);
     }
 
+    @SuppressWarnings({"unchecked"})
```

Comment:
I don't believe we need this here?

Reply:
Turns out my IDE was lying - ignore me!

Reply:
Ah, I see what the root problem is here. What we actually want to do is edit `ESTestCase.randomFrom` as follows. Then you won't need to add `@SuppressWarnings({"unchecked"})` to test methods that call `randomFrom()`.

```diff
diff --git a/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java b/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
index a18fd0d6bcd..991550fe334 100644
--- a/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java
@@ -713,6 +713,8 @@ public abstract class ESTestCase extends LuceneTestCase {
     }
 
     /** Pick a random object from the given array. The array must not be empty. */
+    @SafeVarargs
+    @SuppressWarnings("varargs")
     public static <T> T randomFrom(T... array) {
         return randomFrom(random(), array);
     }
```

Reply:
Grat suggestion! Significantly reduced number of files changed.